src/umpire/resource/NUMAMemoryResourceFactory.cpp

```diff
@@ -0,0 +1,66 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/resource/NUMAMemoryResourceFactory.hpp"
+
+#include "umpire/resource/NUMAMemoryResource.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+#if defined(UMPIRE_ENABLE_NUMA)
+#include <numa.h>
+#endif
+
+namespace umpire {
+namespace resource {
+
+NUMAMemoryResourceFactory::NUMAMemoryResourceFactory(const int numa_node_)
+  : numa_node(numa_node_) {}
+
+std::size_t
+NUMAMemoryResourceFactory::getNodeCount() {
+#if defined(UMPIRE_ENABLE_NUMA)
+  if (numa_available() < 0) UMPIRE_ERROR("libnuma is unusable.");
+  return numa_max_possible_node() + 1;
+#else
+  return 0;
+#endif
+}
+
+bool
+NUMAMemoryResourceFactory::isValidMemoryResourceFor(const std::string& UMPIRE_UNUSED_ARG(name),
+                                                    const MemoryResourceTraits traits)
+  noexcept
+{
+  return (traits.numa_node == numa_node);
+}
+
+std::shared_ptr<MemoryResource>
+NUMAMemoryResourceFactory::create(const std::string& UMPIRE_UNUSED_ARG(name), int id)
+{
+  MemoryResourceTraits traits;
+
+  traits.unified = false;
+  traits.size = 0;
+  traits.numa_node = numa_node;
+
+  traits.vendor = MemoryResourceTraits::vendor_type::IBM;
+  traits.kind = MemoryResourceTraits::memory_type::DDR;
```

Comment:
This is not always the case, especially if the NUMA node is on a GPU....

Reply:
I did this because I copied the HostResourceFactory method, which uses IBM. I'll go back and change both of these.